mailpoet/assets/js/src/automation/editor/components/header/index.tsx

```diff
@@ -57,6 +59,43 @@ function SaveDraftButton(): JSX.Element {
   );
 }
 
+function DeactivateButton(): JSX.Element {
+  const [showDeactivateModal, setShowDeactivateModal] = useState(false);
+  const [isBusy, setIsBusy] = useState(false);
+  const { hasUsersInProgress } = useSelect(
+    (select) => ({
+      hasUsersInProgress:
+        select(storeName).getWorkflowData().stats.totals.in_progress > 0,
+    }),
+    [],
+  );
+
+  const toggleModal = () => {
+    setShowDeactivateModal(!showDeactivateModal);
+  };
```

Comment:
I slightly fear using `toggle` logic in these cases, in case something runs for any random reason twice.

In other words, I think the actions you have here are only "open modal" (the button) and close modal (via `onClose`). There's no reason that the `Deactivate` button actually toggles the modal because a click outside of it will close it anyway.

Reply:
Addressed in [beee962](https://github.com/mailpoet/mailpoet/pull/4450/commits/beee9625396475455b25809e287124de51967a15)